Firestore/core/test/firebase/firestore/model/field_value_test.cc

```diff
@@ -216,14 +216,14 @@ TEST(FieldValue, Copy) {
   clone = null_value;
   EXPECT_EQ(FieldValue::NullValue(), clone);
   EXPECT_EQ(FieldValue::NullValue(), null_value);
-  clone = clone;
+  clone = *&clone;
```

Comment:
Can you split out this change into a separate PR (and add Zhi as a reviewer on that)? It would also be great to add a (single) brief comment with a link to Clang docs which explain the issue.